airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/split_secrets/JsonSecretsProcessorFactory.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.config.persistence.split_secrets;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.common.base.Preconditions;
+import io.airbyte.commons.json.Jsons;
+import lombok.Builder;
+
+@Builder
+public class JsonSecretsProcessorFactory {
+
+  @Builder.Default
+  private boolean maskSecrets = true;
+
+  @Builder.Default
+  private boolean copySecrets = true;
+
+  public JsonSecretsProcessor createJsonSecretsProcessor() {
+    return new JsonSecretsProcessor() {
+
+      @Override
+      public JsonNode maskSecrets(final JsonNode obj, final JsonNode schema) {
+        if (maskSecrets) {
+          // if schema is an object and has a properties field
+          if (!canBeProcessed(schema)) {
+            return obj;
+          }
+          Preconditions.checkArgument(schema.isObject());
+
+          final SecretKeys secretKeys = getAllSecretKeys(schema);
+          return maskAllSecrets(obj, secretKeys);
+        }
+
+        return obj;
+      }
+
+      @Override
+      public JsonNode copySecrets(final JsonNode src, final JsonNode dst, final JsonNode schema) {
+        if (copySecrets) {
+          if (!canBeProcessed(schema)) {
+            return dst;
+          }
+          Preconditions.checkArgument(dst.isObject());
+          Preconditions.checkArgument(src.isObject());
+
+          final ObjectNode dstCopy = dst.deepCopy();
```

Comment:
It seems to be out of the scope of this PR. It will potentially be breaking the logic here. The goal of the PR is really to address the folllow up of the previous PR. I realized that I forget to update the description to make it explicit.